sympy/matrices/matrices.py

```diff
@@ -26,6 +26,9 @@
 
 def _iszero(x):
     """Returns True if x is zero."""
+    if len(x.free_symbols) == 0:
```

Comment:
The len function call and comparison with zero are extraneous here. This is because x.free_symbols returns a python set data type. Python allows direct testing of container data types where an empty container returns false and is otherwise true. 

In this case you could use
`if not x.free_symbols:`
(the "not" is included so that it will return `True` when the set is empty)

This reduces the complexity of the statement.